js/ui/appDisplay.js

```diff
@@ -647,19 +650,29 @@ const AllView = new Lang.Class({
                 this.addItem(icon);
         }));
 
-        // Add the App Center icon if it is enabled and installed
-        if (global.settings.get_boolean(EOS_ENABLE_APP_CENTER_KEY)) {
-            let app = appSys.lookup_app(EOS_APP_CENTER_ID);
-            if (app)
-                this.addItem(new AppCenterIcon(app));
-            else
-                log('App center ' + EOS_APP_CENTER_ID + ' is not installed');
-        }
-
+        // Add the App Center icon if it is enabled (and installed)
+        this._maybeAddAppCenterIcon();
 
         this.loadGrid();
     },
 
+    _maybeAddAppCenterIcon: function() {
+        if (this._appCenterIcon)
+            return;
+
+        if (!global.settings.get_boolean(EOS_ENABLE_APP_CENTER_KEY))
+            return;
+
+        let appSys = Shell.AppSystem.get_default();
+        if (!appSys.lookup_app(EOS_APP_CENTER_ID)) {
+            log('App center ' + EOS_APP_CENTER_ID + ' is not installed');
```

Comment:
(and yes, disabling the icon grid by installing `setup-shell.json` would probably have avoided the crash too, but I think including known-crashing code paths that happen not to be used is strictly worse than logging an error and continuing!